install_test_visibility.sh

```diff
@@ -234,6 +242,141 @@ install_dotnet_tracer() {
   echo "DD_TRACER_VERSION_DOTNET=$(dotnet tool list --tool-path $ARTIFACTS_FOLDER | grep dd-trace | awk '{print $2}')"
 }
 
+is_ruby_version_compliant() {
+  local ruby_version
+  ruby_version=$(ruby -v | cut -d ' ' -f 2)
+
+  local major_ruby_version
+  local minor_ruby_version
+
+  major_ruby_version=$(extract_major_version $ruby_version)
+  minor_ruby_version=$(extract_minor_version $ruby_version)
+
+  if [ "$major_ruby_version" -lt 2 ] || [ "$major_ruby_version" -eq 2 ] && [ "$minor_ruby_version" -lt 7 ]; then
+    return 1
+  fi
+}
+
+is_rubygems_version_compliant() {
+  local rubygems_version
+  rubygems_version=$(gem -v)
+
+  local major_rubygems_version
+  local minor_rubygems_version
+
+  major_rubygems_version=$(extract_major_version $rubygems_version)
+  minor_rubygems_version=$(extract_minor_version $rubygems_version)
+
+  if [ "$major_rubygems_version" -lt 3 ] || [ "$major_rubygems_version" -eq 3 ] && [ "$minor_rubygems_version" -lt 3 ]; then
+    return 1
+  fi
+}
+
+is_gem_present() {
+  if ! bundle info $1 >/dev/null 2>&1 ; then
+    return 1
+  fi
+}
+
+is_gem_datadog_version_compliant() {
+  # if there is no datadog gem in the bundle, it's ok, we are going to add it
+  if ! is_gem_present "datadog"; then
+    return 0
+  fi
+
+  local datadog_version
+  datadog_version=$(bundle info datadog | head -n 1 | awk -F '[()]' '{print $2}')
+
+  local major_datadog_version
+  local minor_datadog_version
+
+  major_datadog_version=$(extract_major_version $datadog_version)
+  minor_datadog_version=$(extract_minor_version $datadog_version)
+
+  if [ "$major_datadog_version" -eq 2 ] && [ "$minor_datadog_version" -lt 4 ]; then
+    return 1
+  fi
+}
+
+is_datadog_ci_version_compliant() {
+  if ! is_gem_present "datadog-ci"; then
+    return 1
+  fi
+
+  local datadog_ci_version
+  datadog_ci_version=$(bundle info datadog-ci | head -n 1 | awk -F '[()]' '{print $2}')
+
+  local major_datadog_ci_version
+  local minor_datadog_ci_version
+
+  major_datadog_ci_version=$(extract_major_version $datadog_ci_version)
+  minor_datadog_ci_version=$(extract_minor_version $datadog_ci_version)
+
+  if [ "$major_datadog_ci_version" -lt 1 ] || [ "$major_datadog_ci_version" -eq 1 ] && [ "$minor_datadog_ci_version" -lt 9 ]; then
```

Comment:
I think you need to wrap those in parentheses, otherwise they don't have the expected precedence in bash:

```
$ major_datadog_ci_version=0
$ minor_datadog_ci_version=10

$ if [ "$major_datadog_ci_version" -lt 1 ] || [ "$major_datadog_ci_version" -eq 1 ] && [ "$minor_datadog_ci_version" -lt 9 ]; then echo "hey"; fi

$ if [ "$major_datadog_ci_version" -lt 1 ] || ( [ "$major_datadog_ci_version" -eq 1 ] && [ "$minor_datadog_ci_version" -lt 9 ] ); then echo "hey"; fi
hey
```


```suggestion
  if [ "$major_datadog_ci_version" -lt 1 ] || ( [ "$major_datadog_ci_version" -eq 1 ] && [ "$minor_datadog_ci_version" -lt 9 ] ); then
```

Reply:
This applies to the other similar tests in the script.

Reply:
fixed in https://github.com/DataDog/test-visibility-install-script/pull/12/commits/e57b098d57bd8bffb6f6ef70bf8248cfe31620ed